### DISCUSSION TITLE
while resolving dids this warning occurs in veramo for react-native

### DISCUSSION THREAD

**Discussion by Saurabhs-333-hub** [decentralized-identity/veramo#1188] [Q&A]
using reactnative.. i have successfully setup and got it running
first time it works.. When I try creating again, its giving this error
I have just followed the steps here - [https://veramo.io/docs/react_native_tutorials/react_native_1_setup_identifiers]
[](https://veramo.io/docs/react_native_tutorials/react_native_1_setup_identifiers)
when i try again with the same DB_ENCRYPTION_KEY it is giving me error like this
is it the error of DB_ENCRYPTION_KEY or any other reason is for this error please resolve it or give some suggetions for this problem

there are screenshots of errors:
![WhatsApp Image 2023-06-15 at 10 41 38 PM (1)](https://github.com/uport-project/veramo/assets/90748697/157f800a-5916-4f58-bd42-3ac1b78fe7b7)
![WhatsApp Image 2023-06-15 at 10 41 38 PM](https://github.com/uport-project/veramo/assets/90748697/e60ca7d4-591f-4328-a681-ad68f414df1e)
![Screenshot (10)](https://github.com/uport-project/veramo/assets/90748697/a9f30d9c-5c6b-491f-a322-45804a9c017f)

**Comment by simonas-notcat** [comment]
Can you share your agent config and the code that gives you this error?

**Reply by Saurabhs-333-hub** [reply]
// filename: setup.ts

// imports:
// Core interfaces
var rand = require("random-key");
import { createAgent, IDataStore, IDataStoreORM, IDIDManager, IKeyManager, IResolver } from '@veramo/core'

// Core identity manager plugin. This allows you to create and manage DIDs by orchestrating different DID provider packages.
// This implements `IDIDManager`
import { DIDManager } from '@veramo/did-manager'

// Core key manager plugin. DIDs use keys and this key manager is required to know how to work with them.
// This implements `IKeyManager`
import { KeyManager } from '@veramo/key-manager'

// This plugin allows us to create and manage `did:ethr` DIDs. (used by DIDManager)
import { EthrDIDProvider } from '@veramo/did-provider-ethr'

// A key management system that uses a local database to store keys (used by KeyManager)
import { KeyManagementSystem, SecretBox } from '@veramo/kms-local'

// Storage plugin using TypeORM to link to a database
import { Entities, KeyStore, DIDStore, migrations, PrivateKeyStore } from '@veramo/data-store'

// TypeORM is installed with '@veramo/data-store'
import { DataSource } from 'typeorm'

// filename: setup.ts

// ... imports

// CONSTANTS
// You will need to get a project ID from infura https://www.infura.io
const INFURA_PROJECT_ID = '68175680........my infura project id is here'

// This is a raw X25519 private key, provided as an example.
// You can run `npx @veramo/cli config create-secret-key` in a terminal to generate a new key.
// In a production app, this MUST NOT be hardcoded in your source code.
var key= rand.generate();
console.log(key)
const DB_ENCRYPTION_KEY = '29739248cad1bd1a0fc4d9b75cd4d2990de675baf5caadfdf8d8f86664aa830c'

// filename: setup.ts

// ... imports & CONSTANTS

// DB setup:
let dbConnection = new DataSource({
  type: 'expo',
  driver: require('expo-sqlite'),
  database: 'veramo.sqlite',
  migrations: migrations,
  migrationsRun: true,
  logging: ['error', 'info', 'warn'],
  entities: Entities,
}).initialize()

// filename: src/veramo/setup.ts

// ... imports & CONSTANTS & DB setup

// Veramo agent setup
export const agent = createAgent<IDIDManager & IKeyManager & IDataStore & IDataStoreORM>({
  plugins: [
    new KeyManager({
      store: new KeyStore(dbConnection),
      kms: {
        local: new KeyManagementSystem(new PrivateKeyStore(dbConnection, new SecretBox(DB_ENCRYPTION_KEY))),
      },
    }),
    new DIDManager({
      store: new DIDStore(dbConnection),
      defaultProvider: 'did:ethr:goerli',
      providers: {
        'did:ethr:goerli': new EthrDIDProvider({
          defaultKms: 'local',
          network: 'goerli',
          name: 'goerli',
          rpcUrl: 'https://goerli.infura.io/v3/' + INFURA_PROJECT_ID,
          gas: 1000001,
          ttl: 31104001,
        }),
      },
    }),
  ],
})

**Reply by Saurabhs-333-hub** [reply]
// it is additional , i 'm just trying to generate the new random id but i haven't used it in the code 
var key= rand.generate();
console.log(key)
i'm just printing it in console

**Reply by Saurabhs-333-hub** [reply]
i have also tried it
npx @veramo/cli config create-secret-key

X25519 raw private key (hex encoded):

        f86c76b1278eb0334b2160da42cb171b4061cf16e4c5375b67cd53178566f7e5

        You can use this key with @veramo/kms-local#SecretBox
        or replace the default agent.yml 'dbEncryptionKey' constant

i replaced the dbencryptionkey with this  f86c76b1278eb0334b2160da42cb171b4061cf16e4c5375b67cd53178566f7e5 but it now continuously showing same errors and not generating did anymore

**Reply by Saurabhs-333-hub** [reply]
@simonas-notcat please check it
